app/models/configuration_script.rb

```diff
@@ -1,3 +1,17 @@
 class ConfigurationScript < ActiveRecord::Base
-  belongs_to :configuration_manager, :class_name => 'ManageIQ::Providers::ConfigurationManager'
+  belongs_to  :manager,
+              :class_name  => 'ManageIQ::Providers::ConfigurationManager',
+              :foreign_key => :configuration_manager_id
+  include ProviderObjectMixin
+
+  def run(vars=nil)
+    extra_vars = vars.reverse_merge(variables)
```

Comment:
@mkanoor - The Tower API will only accept extra_vars (variables) in JSON.  e.g. We send them over to the gem in a HASH via this format - {'extra_vars' => <JSON string>} 

That being said there is a bug in the gem: It is currently only passing on the extra_vars as a JSON string - We need to pass it off to the API in this fashion: {"extra_vars": <JSON string>}. https://github.com/ManageIQ/ansible_tower_client/pull/15 - handled that issue but needs to be implemented a better way - so was closed.